portfolio/src/main/java/com/google/sps/auth/LoggedInState.java

```diff
@@ -0,0 +1,45 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.auth;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.sps.data.AuthStatus;
+
+/** Active state when a user is logged in. */
+public class LoggedInState extends AuthState {
```

Comment:
I think you can combine LoggedInState and LoggedOutState to simplify your code. There really isn't a difference between them, and you already know if the user is logged in or out via isLoggedIn().

Reply:
I can combine them into a single class. getStatus returns different because it creates a different URL based on if the user is logged in or not.

Reply:
```
this.isLoggedIn() ? userService.createLogoutURL(this.redirectUrl)) : userService.createLoginURL(this.redirectUrl));
```